em-client/src/models.rs

```diff
@@ -919,7 +919,7 @@ impl ApplicationConfigExtra {
 
 
 /// 
-#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize, Hash)]
```

Comment:
You shouldn't need this. You need to calculate the cryptographic hash before parsing the string